examples/mqtt-paho/main.c

```diff
@@ -0,0 +1,228 @@
+#define IS_CLEAN_SESSION        1
+#define NOT_RETAINED_MSG        0
+
+static MQTTClient client;
+static Network network;
+static unsigned char buf[BUF_SIZE];
+static unsigned char readbuf[BUF_SIZE];
+
+
+static unsigned get_qos(const char *str)
+{
+    int qos = atoi(str);
+
+    switch (qos) {
+        case 1:     return QOS1;
+        case 2:     return QOS2;
+        default:    return QOS0;
+    }
+}
+
+static int mqtt_launch_thread(void)
+{
+    printf("Launching MQTT Task\n");
+    return MQTTStartTask(&client);
+}
+
+static void _on_msg_received(MessageData *data)
+{
+    printf("mqtt_paho_example: message received on topic"
+           " %.*s: %.*s\n\n",
+           (int)data->topicName->lenstring.len,
+           data->topicName->lenstring.data, (int)data->message->payloadlen,
+           (char *)data->message->payload);
+}
+
+static int cmd_con(int argc, char **argv)
+{
+    if (argc < 7) {
+        printf(
+            "usage: %s <ipv6 addr> <port> <clientid> <user> <password> <KeepAliveInterval in sec>\n",
+            argv[0]);
+        return 1;
+    }
+
+    char *remote_ip = argv[1];
+    int port = atoi(argv[2]);
+
+    MQTTNetworkInit(&network);
+    printf("Trying to connect to %s , port: %d\n",
+           remote_ip, port);
+    if (MQTTNetworkConnect(&network, remote_ip, port)) {
+        printf("error: Unable to connect to %s:%d\n", remote_ip,
+               port);
+        return 1;
+    }
+
+    MQTTClientInit(&client, &network, COMMAND_TIMEOUT_MS, buf, BUF_SIZE, readbuf,
+                   BUF_SIZE);
+
+    MQTTPacket_connectData data = MQTTPacket_connectData_initializer;
+    data.MQTTVersion = MQTT_VERSION;
+    data.clientID.cstring = argv[3];
+    data.username.cstring = argv[4];
+    data.password.cstring = argv[5];
+    data.keepAliveInterval = atoi(argv[6]);
+    data.cleansession = IS_CLEAN_SESSION;
+    printf("Connecting to %s %d\n", remote_ip, port);
+
+    int ret = MQTTConnect(&client, &data);
+    if (ret) {
+        printf("error: Unable to connect client %d\n", ret);
+    } else {
+        printf("Connection successfully\n");
+    }
+    return ret;
+}
+
+static int cmd_discon(int argc, char **argv)
+{
+    (void)argc;
+    (void)argv;
+
+    int res = MQTTDisconnect(&client);
+    if (res) {
+        puts("error: Unable to disconnect");
+    }
+    else {
+        puts("Disconnect successful");
+    }
+
+    MQTTNetworkDisconnect(&network);
+    return res;
+}
+
+static int cmd_pub(int argc, char **argv)
+{
+    enum QoS qos = QOS0;
+
+    if (argc < 3) {
+        printf("usage: %s <topic name> <string msg> [QoS level]\n",
+               argv[0]);
+        return 1;
+    }
+    if (argc == 4) {
+        qos = get_qos(argv[3]);
+    }
+    MQTTMessage message;
+    message.qos = qos;
+    message.retained = NOT_RETAINED_MSG;
+    message.payload = argv[2];
+    message.payloadlen = strlen(message.payload);
+
+    int rc;
+    if ((rc = MQTTPublish(&client, argv[1], &message)) != 0) {
```

Comment:
In the [FreeRTOS sample upstream](https://github.com/eclipse/paho.mqtt.embedded-c/blob/29ab2aa29c5e47794284376d7f8386cfd54c3eed/MQTTClient-C/samples/FreeRTOS/MQTTEcho.c), there's a call to `MQTTYield` after a publish (see [here](https://github.com/eclipse/paho.mqtt.embedded-c/blob/29ab2aa29c5e47794284376d7f8386cfd54c3eed/MQTTClient-C/samples/FreeRTOS/MQTTEcho.c#L73)). Do you think it would make sense to add such a call there as well ?

Reply:
In our case MQTT_TASK is defined, and there's a #if !defined(MQTT_TASK) just before the Yield.

Reply:
Indeed, I misread the _not_ defined...

Reply:
:)